database/misc/mongodb_replset.py

```diff
@@ -0,0 +1,227 @@
+    pymongo_found = False
+
+
+def read_mongocnf_creds():
+    """ Read credentials from ~/.mongodb.cnf file, when exists.
+
+    :return: tuple of username and password
+    """
+    config = ConfigParser.RawConfigParser()
+    try:
+        config.read(os.path.expanduser('~/.mongodb.cnf'))
+        return (config.get('client', 'user'), config.get('client', 'pass'))
+    except (ConfigParser.Error, IOError):
+        return (None, None)
+
+
+def replset_conf(client):
+    """ Return replica set configuration; the same as rs.conf() in shell.
+
+    :param client: initialized Mongo client
+    :return: replica set config
+    """
+    return client['local'].system.replset.find_one()
+
+
+def add_members(client, members):
+    """ Add new members to the replica set.
+
+    :param client: initialized Mongo client
+    :param members: list of tuples that defines hostnames and ports of the
+                    replica set members;
+                    example: `[(mango0, 27017), (mango1, 27018)]`
+    """
+    conf = replset_conf(client)
+    curr_hosts = [m['host'] for m in conf['members']]
+    new_hosts = format_hosts(members)
+
+    new_id = max([int(x['_id']) for x in conf['members']]) + 1
+    for host in (set(new_hosts) - set(curr_hosts)):
+        conf['members'] += [{'_id': new_id, 'host': host}]
+        new_id += 1
+
+    conf['version'] += 1
+    client.admin.command('replSetReconfig', conf)
+
+
+def members_state(client):
+    """ Return dict of the replica set members with their state info. """
+    members = client.admin.command('replSetGetStatus')['members']
+    return dict((m['name'], {'state': m['stateStr']}) for m in members)
+
+
+def replset_initiate(client, name, members):
+    """ Initiate replica set with the specified members.
+
+    :param client: initialized Mongo client
+    :param name: name of the replica set to initiate
+    :param members: list of tuples that defines hostnames and ports of the
+                    replica set members;
+                    example: `[(mango0, 27017), (mango1, 27018)]`
+    """
+    hosts = [{'_id': idx, 'host': join_colon(val)} for idx, val in enumerate(members)]
+    conf = {'_id': name, 'members': hosts}
+    client.admin.command('replSetInitiate', conf)
+
+
+def format_hosts(members):
+    """
+    :param members: list of tuples that contain a hostname and a port
+    :return: list of strings of the form `hostname[:port]`
+    """
+    return [join_colon(x) for x in members]
+
+
+def join_colon(iterable):
+    return ':'.join(str(s) for s in iterable)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec={
+            'login_user':     {'aliases': ['user']},
+            'login_password': {'aliases': ['password'], no_log: True},
+            'login_host':     {'aliases': ['host'], 'default': 'localhost'},
+            'login_port':     {'aliases': ['port'], 'default': 27017},
+            'hosts':          {'aliases': ['members'], 'required': True},
```

Comment:
ansible has some magic included to handle lists, no need to pass a comma separated list and split it: set `type='list'` and I would also add an alias `host`, this way you use it like
~~~
tasks:
- mongodb_replset
  host: your_host
~~~
as well as
~~~
tasks:
- mongodb_replset
  hosts: 
  - your_host
  - your_2nd_host
~~~
and also as comma separated list as well...